src/ShellClients/PanelWindow.vala

```diff
@@ -152,16 +146,13 @@ public class Gala.PanelWindow : Object {
     }
 
     public void set_hide_mode (Pantheon.Desktop.HideMode hide_mode) {
-        clone.hide_mode = hide_mode;
-
-        destroy_barrier ();
-
         if (hide_mode == NEVER) {
             make_exclusive ();
         } else {
             unmake_exclusive ();
-            setup_barrier ();
         }
+
+        clone.hide_mode = hide_mode;
```

Comment:
Another thing I just noticed: Any reason this was moved down? That causes that struts don't update when changing to NEVER.

Reply:
Oops, I forgot to push the fix